File: 2015/Python/src/day_02.py
from pathlib import Path

with open(Path(__file__).parent.parent / "inputs/day_02.txt") as f:
    file = f.read().strip().splitlines()


def part_1() -> None:
    result = 0
    for i in file:
        l, w, h = [int(n) for n in i.split("x")]
        x, y, z = l * w, w * h, h * l
        result += (2 * x + 2 * y + 2 * z) + min(x, y, z)

    print(f"Day 2, Part 1: {result}")


def part_2() -> None:
    result = 0
    for i in file:
        l, w, h = size = [int(n) for n in i.split("x")]
        size.remove(max(size))
        x, y = size
        result += (2 * x + 2 * y) + (l * w * h)

    print(f"Day 2, Part 2: {result}")


if __name__ == "__main__":
    part_1()
    part_2()

File: 2023/Go/src/02/day_02.go
package main

import (
	"fmt"
	"log"
	"os"
	"strings"
	"time"
)

func parse(path string) *[]string {
	file, err := os.ReadFile(path)
	if err != nil {
		log.Fatalf("Error reading file %v", err)
	}
	stripped := strings.TrimSpace(string(file))
	lines := strings.Split(stripped, "\n")

	return &lines
}

func part1(lines *[]string) {
	result := 0
	fmt.Printf("Day 0: Part 1: %v\n", result)
}

func part2(lines *[]string) {
	result := 0
	fmt.Printf("Day 0: Part 2: %v\n", result)
}

func main() {
	startTime := time.Now()
	lines := parse("inputs/day_02.txt")
	fmt.Printf("Finished parsing in %v\n", time.Since(startTime))
	part1(lines)
	fmt.Printf("Finished Part 1 in %v\n", time.Since(startTime))
	part2(lines)
	fmt.Printf("Finished Part 2 in %v\n", time.Since(startTime))
}
